state/keys_test.go

```diff
@@ -84,3 +84,50 @@ func TestMalformedKey(t *testing.T) {
 	keys := make(Keys)
 	require.False(keys.Add("", Read))
 }
+
+func TestHasPermissions(t *testing.T) {
```

Comment:
if we're going to add unit tests here, let's add them for all of the relevant cases

Reply:
Done. I added at least 2 unit tests for each variant

Reply:
Could we add test cases for each of Read/Write/Allocate/All/None has/has not each of the individual permissions ie. read/write/allocate?

Reply:
Instead used sets to check if the perm of the current subtest is contained in all the perms. If yes, we check that it is what we expected, the same is true for the opposite.